plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ContractDescriptionConversionVisitor.kt

```diff
@@ -174,8 +180,10 @@ class ContractDescriptionConversionVisitor(
         data: ContractVisitorContext,
     ): ExpEmbedding {
         val argVar = isInstancePredicate.arg.embeddedVar()
-        val subtypeRel = Is(argVar, ctx.embedType(isInstancePredicate.type))
-        return if (isInstancePredicate.isNegated) Not(subtypeRel) else subtypeRel
+        val argSymbol = isInstancePredicate.arg.getTargetParameter(data)
+        val role = SourceRole.IsTypeCondition(argSymbol, isInstancePredicate.type, isInstancePredicate.isNegated)
+        val subtypeRel = Is(argVar, ctx.embedType(isInstancePredicate.type), role)
```

Comment:
This is inaccurate.  When negated, the `Is` by itself does not satisfy the `SourceRole` you give it.